kafka-client/src/main/java/dev/responsive/kafka/api/WindowStoreOptions.java

```diff
@@ -0,0 +1,134 @@
+package dev.responsive.kafka.api;
+
+import static dev.responsive.kafka.store.StoreUtils.durationToMillis;
+
+import java.time.Duration;
+import java.util.Objects;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.streams.processor.StateStore;
+import org.apache.kafka.streams.state.WindowStore;
+
+public class WindowStoreOptions<K, V> {
+
+  // required configs
+  private final String name;
+  private final long windowSize;
+
```

Comment:
can we use the builder pattern here so that we know objects of this class are immutable once they're built?

Reply:
I don't have anything against it, it's just _really_ awkward when mixed with generics (at least to me, maybe a true java expert like @agavra would have a better idea)

That said we only need generic for the key/value serdes, and technically we only need those for the StoreBuilders since that requires the serdes up front, whereas Streams puts the key/value serde setters directly on the Materialized class. It's just down to a really awkward asymmetry in Streams (which I would guess is itself the fallout of some insane serde inheritance logic in the DSL). 

Point is, I was hoping to smooth over that weird asymmetry but  we don't need to, and can use the builder pattern by just moving the serdes to the StoreBuilder parameters. I'll see what that looks like

Reply:
> Point is, I was hoping to smooth over that weird asymmetry but we don't need to, and can use the builder pattern by just moving the serdes to the StoreBuilder parameters. I'll see what that looks like

Make sense. I'll take a trade of a bigger list of parameters for immutability